signer/src/stacks/api.rs

```diff
@@ -241,7 +241,37 @@ pub struct TenureBlocks {
 }
 
 impl TenureBlocks {
-    /// Return a vector of Stacks blocks from the this tenure.
+    /// Create a new one
+    pub fn try_new(blocks: Vec<NakamotoBlock>, info: SortitionInfo) -> Result<Self, Error> {
+        if blocks.is_empty() {
+            return Err(Error::EmptyStacksTenure);
+        }
+        Ok(Self {
+            blocks,
+            anchor_block_hash: info.burn_block_hash.into(),
+            anchor_block_height: info.burn_block_height,
+        })
+    }
+
+    /// Get all the blocks contained in this object.
+    ///
+    /// # Note
+    ///
+    /// The struct need not contain all the blocks in a tenure.
```

Comment:
nit? also below
```suggestion
    /// The struct doesn't need to contain all the blocks in a tenure.
```

Reply:
🙏🏽 updated in https://github.com/stacks-network/sbtc/pull/781/commits/2132edc0310007a2d39b6efa0ac302fa6c1e29e6.